marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/model/CreateAndUpdateModelTest.java

```diff
@@ -207,6 +221,101 @@ private void updateModelEntityTypes() {
         assertEquals("string", loadModel(getHubClient().getFinalClient()).get("definitions").get(MODEL_NAME).get("properties").get(ENTITY_PROPERTY_1).get("datatype").asText());
     }
 
+    private void addFacetableDateTimeProperty() {
+        Assumptions.assumeTrue(isVersionCompatibleWith520Roles());
+
+        String entityTypes = "[\n" +
+            "  {\n" +
+            "    \"entityName\": \"" + MODEL_NAME + "\",\n" +
+            "    \"modelDefinition\": {\n" +
+            "      \"Customer\": {\n" +
+            "        \"required\": [],\n" +
+            "        \"pii\": [\n" +
+            "          \"" + ENTITY_PROPERTY_1 + "\"\n" +
+            "        ],\n" +
+            "        \"elementRangeIndex\": [\n" +
+            "          \"" + ENTITY_PROPERTY_1 + "\"\n" +
+            "        ],\n" +
+            "        \"properties\": {\n" +
+            "          \"" + ENTITY_PROPERTY_1 + "\": {\n" +
+            "            \"datatype\": \"dateTime\",\n" +
+            "            \"facetable\": true,\n" +
+            "            \"collation\": \"http://marklogic.com/collation/codepoint\"\n" +
+            "          }\n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "]";
+
+        controller.updateModelEntityTypes(readJsonArray(entityTypes));
+    }
+
+    private void updateDataType() {
+        Assumptions.assumeTrue(isVersionCompatibleWith520Roles());
+
+        String entityTypes = "[\n" +
+            "  {\n" +
+            "    \"entityName\": \"" + MODEL_NAME + "\",\n" +
```

Comment:
Same thing here - the only thing we want is the the property and elementRangeIndex. And to reduce the duplication across these two requests, you should make a private method that constructs this payload based on a "String type" property, substituting that in for the datatype.